### DISCUSSION TITLE
strip.Show() stops ESP

### DISCUSSION THREAD

**Discussion by linus1999** [Makuna/NeoPixelBus#386] [Q&A] [answered]
Hello,

I am using this [https://github.com/256dpi/arduino-mqtt] Library for MQTT handling.
If I call strip.Show() using the DMA Method on an ESP8266, the ESP just stops.
It isn't resetting, but also doesn't continue. Your Example without the MQTT library just works fine.

Any idea, what could cause this problem?

Best regards

**Comment by Makuna** [comment]
Where are you calling the strip.Show() in your MQTT sketch?  It can not be called from within an ISR or a callback within a context of a ISR.

**Reply by linus1999** [reply]
Thank you for moving the topic, fits much better here.
I found the problem, I was initializing the pixelCount with 0 accidentally over MQTT.

The show Method internally calls the method.Update function where a while-loop probably never ends.
Didn't confirm this yet, but I think this is where the problem lies.
Is this behaviour expected/by design?

**Comment by Makuna** [comment, marked as answer]
I have been relying on the general code to handle zero sized data well; no special cases.  In this case, the underlying hardware does not like it and puts it into a state where it doesn't consider the sending of the zero sized packet as finished.  While this specific instance is an Esp8266 DMA issue, it does uncover a design issue that can easily popup down the road.

You can create an issue tracking that Zero sized NeoPixelBus is not well mannered and doesn't provide feedback that it can't support zero.
